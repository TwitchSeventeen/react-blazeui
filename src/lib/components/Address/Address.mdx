---
name: Address
menu: Components
---

import { Playground, PropsTable } from 'docz'
import Address from './index';

# Address

This provides a consistent way of styling addresses

## Basic usage
<Playground>
    <Address>
        10010 The Street,
        <br /> Binary Town,
        <br /> Offandon City,
    </Address>
</Playground>

## Properties
<PropsTable of={Address} />