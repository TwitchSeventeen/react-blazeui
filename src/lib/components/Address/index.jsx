import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Address(props) {
  const {
    children, className, dataName, ...other
  } = props;
  const classes = classNames('c-address', className);
  return (
    <address
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </address>
  );
}

Address.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Address.defaultProps = {
  children: null,
  className: null,
  dataName: 'address',
};

export default Address;
