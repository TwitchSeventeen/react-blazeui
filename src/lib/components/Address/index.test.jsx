import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Address from './index';

describe('Address', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Address).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Address>
        10010 The Street,
        <br />
        {' '}
        Binary Town,
        <br />
        {' '}
        Offandon City,
      </Address>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
