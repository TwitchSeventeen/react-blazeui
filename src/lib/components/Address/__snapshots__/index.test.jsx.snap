// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address renders correctly 1`] = `
<address
  className="c-address"
>
  10010 The Street,
  <br />
   
  Binary Town,
  <br />
   
  Offandon City,
</address>
`;
